### DISCUSSION TITLE
Correct way to use v-model in custom input component

### DISCUSSION THREAD

**Discussion by akira-fukue** [beholdr/maska#103] [Q&A]
In the documentation it says not to use the v-model in masked inputs. I tested it and I always have one more character than the input value.

What's the best way to use maska ​​with custom input components?

**Comment by 25Das** [comment]
In VueJs, with v2.0.1.
I use ` v-model` with `v-maska:[options]`.
And then in `options` I set `{ mask: '#-#', otherOptions... }` and **`onMaska: ({ unmasked }) => (this.foo = unmasked)`** to retrieve value.

**Comment by beholdr** [comment]
Please try v2.1, it should work with custom input components.

**Reply by Disorrder** [reply]
v2.1.2 `data-mask` attribute is not working with custom inputs. It's only possible to set mask with directive's argument like `v-mask:[Time]` where Time is imported object like `{mask: '##:##'}`

**Reply by beholdr** [reply]
@Disorrder possibly because `data-mask` is not passed to `<input>`. Could you provide online sandbox with this case?

**Reply by Disorrder** [reply]
@beholdr Yes, you was right. It works now after passing data attributes to input. Thanks!
